lib/gmail/gmail.go

```diff
@@ -364,6 +364,87 @@ func shardForMsgId(id string) int {
 	return int(shard)
 }
 
+// This function return true if the given label parameter
+// is not an extra (decorating) label.
+func isToBeAccounted(label string) bool {
+	// The following build a set of the real mails labels.
+	// Only these mails types are to take into account when
+	// retrieving a full mailbox without filtering by a given
+	// label.
+	type void struct{}
```

Comment:
As I noted above, I think I'd just declare a map and access it directly. In addition, given the small size of this map, the optimization of using empty struct members seems to me not worth the slight loss in readability. 

In any event, this function could be rewritten as:

```suggestion
func isCounted(label string) bool {
	_, r := map[string]struct{}{
		"INBOX": struct{}{},
		"DRAFT": struct{}{},
		"SENT":  struct{}{},
		"TRASH": struct{}{}}[label]
	return r
}
```